### PR TITLE
Add a --no-http flag to start command

### PR DESCRIPTION
**Pull request by abarth**
This flag builds a local FLX file and pushes that to the device instead of
using an HTTP server.

### CODE REVIEW

**Comment by abarth** [comment]
@devoncarew 

Sorry this patch got a little large.  I ended up trying to rationalize the difference between host platforms and target platforms in order to differentiate between the sky_snapshot for a Linux host and a sky_snapshot for a Mac host.

File: lib/src/artifacts.dart [hunk 1]
```diff
@@ -10,15 +10,147 @@ import 'dart:io';
...
+      return 'ios-x64';
+    case TargetPlatform.linux:
+      return 'linux-x64';
+  }
+}
+
+// Keep in sync with https://github.com/flutter/engine/blob/master/sky/tools/big_red_button.py#L50
+String _getCloudStorageBaseUrl({String category, String platform, String revision}) {
+  if (platform == 'darwin-x64') {
+    // In the fulness of time, we'll have a consistent URL pattern for all of
```
Comment (devoncarew) [inline lib/src/artifacts.dart]: sp: fullness

File: lib/src/artifacts.dart [hunk 2]
```diff
@@ -31,102 +163,66 @@ class ArtifactStore {
...
-      if (_needsToBeExecutable(artifact)) {
-        ProcessResult result = await Process.run('chmod', ['u+x', cachedFile.path]);
-        if (result.exitCode != 0) throw new Exception(result.stderr);
+    Directory cacheDir = _getCacheDirForArtifact(artifact);
+    File cachedFile = new File(path.join(cacheDir.path, artifact.fileName));
+    if (!cachedFile.existsSync()) {
+      print('Downloading ${artifact.name} from the cloud, one moment please...');
+      await _downloadFile(artifact.getUrl(engineRevision), cachedFile);
+      if (artifact.executable) {
+        ProcessResult result = Process.runSync('chmod', ['u+x', cachedFile.path]);
```
Comment (devoncarew) [inline lib/src/artifacts.dart]: Down the road we should have a `makeExecutable()` utility method, which would have different OS implementations.

File: lib/src/build_configuration.dart [hunk 3]
```diff
@@ -2,29 +2,48 @@
...
+final Logger _logging = new Logger('sky_tools.build_configuration');
+
 enum BuildType {
   prebuilt,
   release,
   debug,
 }
 
-enum BuildPlatform {
+enum HostPlatform {
```
Comment (devoncarew) [inline lib/src/build_configuration.dart]: +1 to the separation into `HostPlatform` and `TargetPlatform`, it makes things clearer.

**Comment by devoncarew** [comment]
lgtm!

File: lib/src/artifacts.dart (hunk 2 above)
Comment (abarth) [inline lib/src/artifacts.dart]: I'll add a TODO

**Pull request merged**